includes/class-amp-travel-taxonomies.php

```diff
@@ -0,0 +1,190 @@
+		add_action( 'edit_activity', array( $this, 'save_activity_svg' ) );
+		add_action( 'create_activity', array( $this, 'save_activity_svg' ) );
+	}
+
+	/**
+	 * Register Activity meta.
+	 */
+	public function register_activity_meta() {
+		$args = array(
+			'sanitize_callback' => array( $this, 'sanitize_activity_svg' ),
+			'type'              => 'string',
+			'single'            => true,
+			'show_in_rest'      => true,
+		);
+		register_meta( 'term', 'amp_travel_activity_svg', $args );
+	}
+
+	/**
+	 * Add metafield to Add Activity form page.
+	 */
+	public function add_activity_meta_fields() {
+		?>
+		<div class='form-field form-required term-svg-wrap'>
+			<label for='travel-activity-svg'>SVG</label>
+			<?php wp_nonce_field( basename( __FILE__ ), 'travel_activity_svg_nonce' ); ?>
+			<textarea aria-required='true' name='travel-activity-svg' id='travel-activity-svg'></textarea>
+			<p class="description"><?php esc_attr_e( 'This is for the background icon of the activity term.', 'travel' ); ?></p>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Sanitize the SVG field.
+	 *
+	 * @param string $text Text.
+	 * @return string
+	 */
+	public function sanitize_activity_svg( $text ) {
+		$allowed_html = array(
+			'path' => array(
+				'fill' => true,
+				'd'    => true,
+			),
+			'svg'  => array(
+				'class'   => true,
+				'viewbox' => true,
+			),
+		);
+
+		// Replace ' with " since ' might cause issues with sanitizing.
+		$text = trim( str_replace( "'", '"', $text ) );
+		return wp_kses( $text, $allowed_html );
+	}
+
+	/**
+	 * Add SVG field to Activity term edit view.
+	 *
+	 * @param WP_Term $term Term object.
+	 */
+	public function edit_activity_meta_fields( $term ) {
+		$value = get_term_meta( $term->term_id, 'amp_travel_activity_svg', true );
+		if ( empty( $value ) ) {
+			$value = '';
+		}
+		?>
+		<tr class="form-field term-svg-wrap">
+			<th scope="row"><label for="travel-activity-svg"><?php esc_attr_e( 'SVG', 'travel' ); ?></label></th>
+			<td>
+				<?php wp_nonce_field( basename( __FILE__ ), 'travel_activity_svg_nonce' ); ?>
+				<textarea aria-required="true" name="travel-activity-svg" id="travel-activity-svg">
+				<?php
+				// @codingStandardsIgnoreLine
+				echo $this->sanitize_activity_svg( $value );
+				?>
+			</textarea>
+				<p class="description"><?php esc_attr_e( 'This is for the background icon of the activity term.', 'travel' ); ?></p>
+			</td>
+		</tr>
+		<?php
+	}
+
+	/**
+	 * Save Activity SVG value on adding and editing the term.
+	 *
+	 * @param integer $term_id Term ID.
+	 */
+	public function save_activity_svg( $term_id ) {
+
+		if ( ! wp_verify_nonce( $_POST['travel_activity_svg_nonce'], basename( __FILE__ ) ) ) {
+			return;
+		}
+
+		$old_value = get_term_meta( $term_id, 'amp_travel_activity_svg', true );
+		$new_value = isset( $_POST['travel-activity-svg'] ) ? $this->sanitize_activity_svg( $_POST['travel-activity-svg'] ) : '';
+		if ( $old_value !== $new_value ) {
```

Comment:
Good question, thought so initially. For example if the `svg` tag has illegal attributes then the `svg` tag will still remain, in theory it would get stripped entirely only if none of the inserted tags are allowed. But perhaps it makes more sense if the new value is saved as empty only if the unstripped version is empty (the intention was to save empty value). Will change it for now.